src/Nethermind/Nethermind.Synchronization/SyncServer.cs

```diff
@@ -192,25 +192,24 @@ public void AddNewBlock(Block block, ISyncPeer nodeWhoSentTheBlock)
                     if (_logger.IsInfo) _logger.Info($"Peer {nodeWhoSentTheBlock} sent block {block} with total difficulty {block.TotalDifficulty} higher than TTD {_specProvider.TerminalTotalDifficulty}");
                 }
 
-                if (!ValidateSeal(block, nodeWhoSentTheBlock))
-                {
-                    ThrowOnInvalidBlock(block, nodeWhoSentTheBlock);
-                }
-
-                bool isKnownParent = block.ParentHash is not null && _blockTree.IsKnownBlock(block.Number - 1, block.ParentHash);
-                if (isKnownParent)
+                Block? parent = _blockTree.FindBlock(block.ParentHash);
+                if (parent != null)
                 {
                     // we null total difficulty for a block in a block tree as we don't trust the message
                     UInt256? totalDifficulty = block.TotalDifficulty;
 
-                    // we do not trust total difficulty from peers
-                    // Parity sends invalid data here and it is equally expensive to validate and to set from null
-                    block.Header.TotalDifficulty = null;
+                    // Recalculate total difficulty as we don't trust total difficulty from gossip
+                    block.Header.TotalDifficulty = parent.TotalDifficulty + block.Header.Difficulty;
                     if (!_blockValidator.ValidateSuggestedBlock(block))
                     {
                         ThrowOnInvalidBlock(block, nodeWhoSentTheBlock);
                     }
 
+                    if (!ValidateSeal(block, nodeWhoSentTheBlock))
```

Comment:
It shouldn't be a problem, but we have to run some nodes. @LukaszRozmej wdyt?

Reply:
I think its fine